core/ipc/twin/grpc/common/src/main/proto/twin-grpc.proto

```diff
@@ -0,0 +1,40 @@
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "org.opennms.core.ipc.twin.grpc.common";
+option java_outer_classname = "TwinGrpcIpc";
+
+// service definitions of Twin IPC between Minion and OpenNMS
+service OpenNMSTwinIpc {
+  // Streams Twin request/Response from Minion to OpenNMS
+  rpc RpcStreaming (stream TwinRequestProto) returns (stream TwinResponseProto) {
+  }
+  // Stream Twin updates from OpenNMS to Minion.
+  rpc SinkStreaming (MinionHeader) returns (stream TwinResponseProto) {
+  }
+}
+/*
+   Twin Request object used by Minion to make RPC Request.
+ */
+message TwinRequestProto {
+  string consumer_key = 1;
+  string system_id = 2;
+  string location = 3;
+  map<string, string> tracing_info = 4;
+}
+
+/*
+    Twin Response object sent by OpenNMS for RPC response as well as for Sink update.
+ */
+message TwinResponseProto {
+  string consumer_key = 1;
+  bytes twin_object = 2;
```

Comment:
Based on this model it looks like we're sending the complete object over the wire every-time there is an update.

Do we plan on changing this API later when we account for incremental updates?

Reply:
yes. Currently API and proto doesn't include version.
Once we include version, we can use same `twin_object` for complete/partial objects.

Reply:
👍 thanks